caas/kubernetes/provider/constraints.go

```diff
@@ -23,5 +27,81 @@ var unsupportedConstraints = []string{
 func (k *kubernetesClient) ConstraintsValidator(ctx envcontext.ProviderCallContext) (constraints.Validator, error) {
 	validator := constraints.NewValidator()
 	validator.RegisterUnsupported(unsupportedConstraints)
-	return validator, nil
+	return &constraintsValidator{Validator: validator}, nil
+}
+
+type constraintsValidator struct {
+	constraints.Validator
+}
+
+// Validate returns an error if the given constraints are not valid, and also
+// any unsupported attributes.
+func (v *constraintsValidator) Validate(cons constraints.Value) ([]string, error) {
+	if cons.Tags != nil {
+
+		var validateTopologyKeyTagFunc = func(key string, topologySpreadKey string, array []string) error {
```

Comment:
`array` is a poor name here

This should just be a private package func like `parseDelimitedValues()` is